src/mds/Server.cc

```diff
@@ -349,10 +349,17 @@ void Server::handle_client_session(MClientSession *m)
 	m->put();
 	return;
       }
+      // This looks as if we are getting a seq that is higher than
+      // expected.  I didn't notice any "wrap" handling, but the seqn is
+      // 64-bits ...
+      //
       if (m->get_seq() != session->get_push_seq()) {
-	dout(0) << "old push seq " << m->get_seq() << " != " << session->get_push_seq() 
+	dout(0) << "old push seq " << m->get_seq() << " != " << session->get_push_seq()
 		<< ", BUGGY!" << dendl;
-	assert(0);
+	mds->clog->warn() << "incorrect push seq " << m->get_seq() << " != "
+			  << session->get_push_seq() << ", dropping0";
```

Comment:
s/dropping0/dropping/

Also, would be nice to include in the log message: session->info.inst.name (the client's unique ID) and session->get_human_name() (a hopefully more recognisable string derived from the metadata), so that the admin has a chance to know which client is buggy.